### DISCUSSION TITLE
npm ERR! gyp verb cli

### DISCUSSION THREAD

**Discussion by velbon** [community/community#47042] [Actions]
### Select Topic Area

Product Feedback

### Body

Good day everyone
Please I need for this issue below:

(base) gerald@gerald-444 platform-client % sudo npm run install:all
Password:

> ushahidi-platform-client@1.0.0 install:all
> npm install && cd api && npm install && cd .. && cd root && npm install && cd .. && cd legacy && npm install && cd .. && cd utilities && npm install && cd ..

up to date, audited 706 packages in 5s

49 packages are looking for funding
  run `npm fund` for details

6 high severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'ushahidi-platform-sdk@0.6.1',
npm WARN EBADENGINE   required: { node: '>= 10 <13' },
npm WARN EBADENGINE   current: { node: 'v18.14.0', npm: '9.3.1' }
npm WARN EBADENGINE }

up to date, audited 1191 packages in 10s

105 packages are looking for funding
  run `npm fund` for details

22 vulnerabilities (2 low, 1 moderate, 17 high, 2 critical)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path /Users/gerald/platform-client/root/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c node scripts/build.js
npm ERR! Building: /usr/local/bin/node /Users/gerald/platform-client/root/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   '/usr/local/bin/node',
npm ERR! gyp verb cli   '/Users/gerald/platform-client/root/node_modules/node-gyp/bin/node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@18.14.0 | darwin | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (/Users/gerald/platform-client/root/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/Users/gerald/platform-client/root/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/Users/gerald/platform-client/root/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:207:21)
npm ERR! gyp verb `which` failed  python2 Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (/Users/gerald/platform-client/root/node_modules/which/which.js:13:12)
npm ERR! gyp verb `which` failed     at F (/Users/gerald/platform-client/root/node_modules/which/which.js:68:19)
npm ERR! gyp verb `which` failed     at E (/Users/gerald/platform-client/root/node_modules/which/which.js:80:29)
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/which/which.js:89:16
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/isexe/index.js:42:5
npm ERR! gyp verb `which` failed     at /Users/gerald/platform-client/root/node_modules/isexe/mode.js:8:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:207:21) {
npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! gyp verb `which` failed }
npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! gyp verb `which` succeeded python /Users/gerald/opt/anaconda3/bin/python
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Command failed: /Users/gerald/opt/anaconda3/bin/python -c import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack   File "<string>", line 1
npm ERR! gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack                       ^
npm ERR! gyp ERR! stack SyntaxError: invalid syntax
npm ERR! gyp ERR! stack
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:419:12)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! gyp ERR! System Darwin 22.2.0
npm ERR! gyp ERR! command "/usr/local/bin/node" "/Users/gerald/platform-client/root/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /Users/gerald/platform-client/root/node_modules/node-sass
npm ERR! gyp ERR! node -v v18.14.0
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok
npm ERR! Build failed with error code: 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gerald/.npm/_logs/2023-02-11T05_40_31_305Z-debug-0.log

**Comment by tarikmardy** [comment]
The error message you're encountering is due to a compatibility issue between the version of Node.js you have installed and the version required by ushahidi-platform-sdk@0.6.1.
The required version is >= 10 <13, while your installed version is v18.14.0.

Downgrade your Node.js version to one that is compatible with the SDK or Upgrade the version of the ushahidi-platform-sdk to a version that is compatible with your installed version of Node.js.

Additionally, there are several deprecated packages that are no longer supported or maintained. It's recommended to upgrade to the latest version or a version that is actively maintained to resolve these warnings.
There are also several vulnerabilities reported by the npm audit. To address these issues, you can run `npm audit fix` to automatically fix issues that do not require attention, or `npm audit fix --force` to address all issues including breaking changes.
